assignment_1/1_permutations.py

```diff
@@ -0,0 +1,44 @@
+#Given two strings, write a method to decide if one is a permutation of the other
+# Magdalena Grodzińska
+
+import re
+from collections import Counter
+
+# helper funtions
+
+def validate_input(input):
+    pattern = '[a-zA-Z]+'
+
+    result = re.fullmatch(pattern, input)
+
+    if result == None:
+        return False
+    else:
+        return True
+
+def check_letters_match(text1, text2):
+    res1 = Counter(text1)
+    res2 = Counter(text2)
+
+    return res1 == res2
+
+# main function
+
+def check_permutation(text1, text2):
+    if validate_input(text1) and validate_input(text2):
```

Comment:
How do you feel about calling validate_input() on both inputs when you consider that check_permutations() is used:
a) directly on user-provided data
b) as a library to be used by other code

Of course, the assignment gave no context to know what type of use to expect and you had no way to immediately ask for context either. My gut feeling is that check_permutation() could document that the inputs have to be strings and omit the validation in the fast path.